pkg/reconciler/labeler/v2/accessors.go

```diff
@@ -51,50 +54,78 @@ var _ Accessor = (*Revision)(nil)
 func NewRevisionAccessor(
 	client clientset.Interface,
 	tracker tracker.Interface,
-	lister listers.RevisionLister) *Revision {
+	lister listers.RevisionLister,
+	clock clock.Clock) *Revision {
 	return &Revision{
 		client:         client,
 		tracker:        tracker,
 		revisionLister: lister,
+		clock:          clock,
 	}
 }
 
 // makeMetadataPatch makes a metadata map to be patched or nil if no changes are needed.
-func makeMetadataPatch(acc kmeta.Accessor, routeName string) (map[string]interface{}, error) {
-	labels, err := addRouteLabel(acc, routeName)
-	if err != nil {
+func makeMetadataPatch(
+	acc kmeta.Accessor, routeName string, addRoutingState bool, clock clock.Clock) (map[string]interface{}, error) {
+	labels := map[string]interface{}{}
+	annotations := map[string]interface{}{}
+
+	if err := addRouteLabel(acc, routeName, labels); err != nil {
 		return nil, err
 	}
 
+	if addRoutingState {
+		markRoutingState(acc, routeName != "", clock, labels, annotations)
+	}
+
+	meta := map[string]interface{}{}
 	if len(labels) > 0 {
-		meta := map[string]interface{}{
-			"metadata": map[string]interface{}{
-				"labels": labels,
-			},
-		}
-		return meta, nil
+		meta["labels"] = labels
+	}
+	if len(annotations) > 0 {
+		meta["annotations"] = annotations
+	}
+	if len(meta) > 0 {
+		return map[string]interface{}{"metadata": meta}, nil
 	}
 	return nil, nil
 }
 
+// markRoutingState updates the RoutingStateLabel and bumps the modified time annotation.
+func markRoutingState(
+	acc kmeta.Accessor, hasRoute bool, clock clock.Clock,
+	diffLabels, diffAnn map[string]interface{}) {
+	var wantState string
+	if hasRoute {
+		wantState = string(v1.RoutingStateActive)
+	} else {
+		wantState = string(v1.RoutingStateReserve)
+	}
```

Comment:
same trick would work here:
```
wantState = string(v1.RoutingStateReserve)
if hasRout {
  wantState = string(v1.RoutingStateActive)
}
```
Or reverse, whichever is going to be more likely.